### ISSUE TITLE
Hardware or software record_generation

### ISSUE THREAD

**Issue by ArendPWS** [BobAtchley/weewx-ws6in1#2]
First of all: thank you for providing this driver, it works very well.

I ran into some issues when using your driver, but it could als be related to the behaviour of WeeWX, I am not sure.

In your readme.txt you write:

> If weewx stops working (due to server problems etc) and there are missing
records, on startup weewx will attempt to restore these from the your weather
station console.  To be successful in the weewx.conf section "[StdArchive]" set
record_generation = hardware (the default is software).

When I look at the original WeeWX-configuration file it reads:

> [StdArchive]
    # If possible, new archive records are downloaded from the station
    # hardware. If the hardware does not support this, then new archive
    # records will be generated in software.
    # Set the following to "software" to force software record generation.
    record_generation = hardware

[https://github.com/weewx/weewx/blob/master/weewx.conf](https://github.com/weewx/weewx/blob/master/weewx.conf)

But it doesn't matter which mode I use, weewx starts downloading the memory contents from the console anyway. It makes no difference which configuration option I am using. I have no idea why my configuration is behaving differently. For testing purposes it can be somewhat annoying that with each restart all memory contents is downloaded each time. Accidentally I found a way to (temporary) stop this downloading behaviour by means of an undocumented weewx option in [StdArchive]. Adding

[StdArchive]
no_catchup = True 

solved it. [https://groups.google.com/g/weewx-user/c/z94fmNeuEIc/m/GRdT09e6AQAJ](https://groups.google.com/g/weewx-user/c/z94fmNeuEIc/m/GRdT09e6AQAJ)

**Comment by BobAtchley** [comment]
I think the behaviour of weewx  has been changed from my early testing (mostly using an early weewx 4.0 beta build) as I had to change record_generation to hardware.
I'll leave this issue open for the moment.  At the next release (none planned in the short term) I'll update the readme.txt with your work around.  The other workaround (if your database is up to date) is to clear the data buffer on the console.  If I'm testing and likely to restart weewx multiple times I backup the database first and then clear the console before starting my testing.
